### DISCUSSION TITLE
What is the purpose of the 'networks' table in version 2.2.0?

### DISCUSSION THREAD

**Discussion by happyjyj** [ory/hydra#3745] [Q&A] [answered]
Hello, I found a new table in hydra v2.2.0, may I know the table's purpose? and all tables have 'nid' field.
<img width="1119" alt="image" src="https://github.com/ory/hydra/assets/99728256/f15a6363-7ff8-4bbe-b78e-9766d0ac0401">

**Comment by glebsts** [comment]
I got related question and was not able to find anything in docs.

**Comment by aeneasr** [comment, marked as answer]
It's an internal value and has currently no real effect in the code base

**Reply by glebsts** [reply]
Well, as I mentioned in slack, it does affect me. Self-hosted 2.1.2 clients created about a year ago have certain `nid`, but in `networks` table there are two networks, one with this `nid`, and second with another (both created timestamp same 2013). And when auth flow happens with client id, `persister` adds `nid` [from context to SQL find-first query](https://github.com/ory/hydra/blob/v2.1.2/persistence/sql/persister_client.go#L25) (`hydra_client.id= <client id> AND hudra_client.nid=<nid from context>`), and starting few weeks ago context for all requests has this second `nid`, not the one that is set on client row in `hydra_client` table for my `<client id>`, so [client is not found and I get error about it](https://github.com/ory/fosite/blob/v0.47.0/authorize_request_handler.go#L345) instead of working auth flow. And if we create new client through CLI/rest, new client has this new `nid` value, the second one. And I cannot understand why it changed in context.
What am I doing wrong?

**Reply by glebsts** [reply]
I ended up with manually deleting "second" network from db are rebooting Ora Hydra pod. Old and valid network ID is now used by context (seems like Ory takes result of `select id from networks limit 1;` and this "second" uuid was `0...` while original was `7...`. 
So, as I still do not usrtstand the reason behind this `0..` appearing (with same creation timestamp as `7...` being `2013-10-07 08:23:19` , and I beieve it is related to [migration that run twice](https://github.com/ory/hydra/blob/v2.1.2/persistence/sql/migrations/20150101000001000000_networks.postgres.up.sql#L20):

Pod cmd is `hydra migrate sql --read-from-env --yes && hydra serve all`. So there should be changelog table to avoid running it twice..
